### PR TITLE
Fix parallel-tsc may throw exception

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: scripts/parallel-tsc/tsc.js [hunk 1]
```diff
@@ -126,14 +126,13 @@ class Pool {
...
-      this.#running.delete(taskProcess);
       await this.#onTaskDone(tasks, code, duration);
-
-      // Spin  the event loop so that #run can pick up the next task
-      await setTimeout(() => {}, 0);
+      this.#running.delete(taskProcess);
 
       this.#runTasks();
 
+      await 0;
```
Comment (liuxingbaoyu) [inline scripts/parallel-tsc/tsc.js:133]: It seems that `setTimeout` is not used here.
Perhaps
```ts
await 0;
await new Promise(resolve => setTimeout(resolve, 0));
```

File: scripts/parallel-tsc/tsc.js [hunk 2]
```diff
@@ -201,7 +201,7 @@ const pool = new Pool({
     done += projects.length;
     if (!process.env.TSCHECK_SILENT) {
       console.log(
-        `[${done}/${total}] ${projects.map(project => "./" + relativePath(process.cwd(), project)).join(" ")} took ${Math.round(duration) / 1e3}s`
+        `[${done}/${total}] ${projects.map(project => relativePath(process.cwd(), project)).join(" ")} took ${Math.round(duration) / 1e3}s`
```
Comment (liuxingbaoyu) [inline scripts/parallel-tsc/tsc.js:204]: On Windows it originally showed `./packages\babel-helper-validator-option`. :)

File: scripts/parallel-tsc/tsc.js [hunk 3]
```diff
@@ -126,11 +126,11 @@ class Pool {
     taskProcess.once("exit", async code => {
       const startTime = this.#running.get(taskProcess);
       const duration = performance.now() - startTime;
-      this.#running.delete(taskProcess);
       await this.#onTaskDone(tasks, code, duration);
+      this.#running.delete(taskProcess);
```
Comment (liuxingbaoyu) [inline scripts/parallel-tsc/tsc.js:130]: Here is the fix.

File: scripts/parallel-tsc/tsc.js [hunk 4]
```diff
@@ -126,11 +126,11 @@ class Pool {
...
 
-      // Spin  the event loop so that #run can pick up the next task
-      await setTimeout(() => {}, 0);
+      // Spin the event loop so that #run can pick up the next task
+      await 0;
```
Comment (nicolo-ribaudo) [inline scripts/parallel-tsc/tsc.js:133]: This doesn't actually spin the event loop. `setTimeout` has lower priority than `queueMicrotask`, while `promise.then` for an already-resolved promise has the same priority.

Comment (liuxingbaoyu) [reply scripts/parallel-tsc/tsc.js:133]: Ah, I see.
Perhaps you meant `await new Promise(resolve => setTimeout(resolve, 0));`?

Comment (nicolo-ribaudo) [reply scripts/parallel-tsc/tsc.js:133]: Oh, yes :sweat_smile: 

Or `import { setTimeout } from "node:timers/promises"`

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**